modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/restclients/OrgMgtRestClient.java

```diff
@@ -0,0 +1,394 @@
+     * Get an M2M access token from the B2B self-service application.
+     *
+     * @return Access token of the B2B self-service application.
+     * @throws Exception If an error occurs while retrieving the access token.
+     */
+    public String getM2MAccessToken() throws Exception {
+
+        OpenIDConnectConfiguration openIDConnectConfiguration =
+                oAuth2RestClient.getOIDCInboundDetails(selfServiceAppId);
+
+        URI tokenEndpoint = new URI(getTenantQualifiedURL(OAuth2Constant.ACCESS_TOKEN_ENDPOINT,
+                tenantInfo.getDomain()));
+        TokenRequest request = getTokenRequest(tokenEndpoint, openIDConnectConfiguration);
+        HTTPResponse tokenHTTPResp = request.toHTTPRequest().send();
+        TokenResponse tokenResponse = TokenResponse.parse(tokenHTTPResp);
+        AccessTokenResponse accessTokenResponse = tokenResponse.toSuccessResponse();
+
+        return accessTokenResponse.getTokens().getAccessToken().getValue();
+    }
+
+    /**
+     * Retrieve a switched M2M token for the given organization.
+     *
+     * @param organizationID ID of the organization.
+     * @return Switched M2M token for the given organization.
+     * @throws Exception If an error occurs while switching the M2M token.
+     */
+    public String switchM2MToken(String organizationID) throws Exception {
+
+        List<NameValuePair> urlParameters = new ArrayList<>();
+        urlParameters.add(new BasicNameValuePair(OAuth2Constant.GRANT_TYPE_NAME, "organization_switch"));
+        urlParameters.add(new BasicNameValuePair("token", getM2MAccessToken()));
+        urlParameters.add(new BasicNameValuePair("scope", "SYSTEM"));
+        urlParameters.add(new BasicNameValuePair("switching_organization", organizationID));
+
+        HttpPost httpPost =
+                new HttpPost(getTenantQualifiedURL(OAuth2Constant.ACCESS_TOKEN_ENDPOINT, tenantInfo.getDomain()));
+        httpPost.setHeader(AUTHORIZATION_ATTRIBUTE, BASIC_AUTHORIZATION_ATTRIBUTE + new String(Base64.encodeBase64(
+                (selfServiceAppClientId + ":" + selfServiceAppClientSecret).getBytes())));
+        httpPost.setHeader(CONTENT_TYPE_ATTRIBUTE, "application/x-www-form-urlencoded");
+        httpPost.setEntity(new UrlEncodedFormEntity(urlParameters));
+
+        HttpResponse response = client.execute(httpPost);
+        String responseBody = EntityUtils.toString(response.getEntity(), "UTF-8");
+        EntityUtils.consume(response.getEntity());
+
+        JSONParser parser = new JSONParser();
+        org.json.simple.JSONObject responseJSONBody = (org.json.simple.JSONObject) parser.parse(responseBody);
+        Assert.assertNotNull(responseJSONBody, "Access token response is null.");
+        Assert.assertNotNull(responseJSONBody.get(OAuth2Constant.ACCESS_TOKEN), "Access token is null.");
+
+        return (String) responseJSONBody.get(OAuth2Constant.ACCESS_TOKEN);
+    }
+
+    private void createB2BSelfServiceApplication(JSONObject authorizedAPIs)
```

Comment:
```suggestion
    private void createB2BApplication(JSONObject authorizedAPIs)
```

Reply:
Resolved with https://github.com/wso2/product-is/pull/20772/commits/1258e75f5c9e9525d851fd5e4143513bfc72946c